doc/features/README.md

```diff
@@ -0,0 +1,64 @@
+Features introspection in e2e tests
+======================================
+
+**Goal**
+
+Provide a way to identify the active features for every supported operator version.
+
+**Design**
+
+- Provide the possibility to group a set of tests that verify a common feature using ginkgo Label("feature:<topic>").
+  A Topic should describe the main feature being tested or the unique effect of the test on the cluster. For instance:
+  `feature:schedrst`: indicates that the test is expected to restart the schedular, be it a pod restart or a complete removal of the scheduler and recreation.
+  `feature:rtetols`: indicates that the test is testing the new feature RTE tolerations, which involves specific non-default configuration on the operator CRs hence functional effects.
+  `feature:unsched`: indicates that the test workload is anticipated to be un-schedulable for any reason for example: insufficient resources (pod is pending), TAE (failed)..
+  `feature:wlplacement`: indicates that the test challenges the scheduler to place a workload on a node. This is a common label for most of the functional tests.
+    - Rules for creating a new topic
+        - A Topic should point out briefly the new feature being tested or the unique effect of the test on the cluster
+        - The topic should consist of small letters
+        - should not consist of any of `&|!,()/`
```

Comment:
let's add a rule for composite words. Is it gonna be `composite_word`? or `composite-word`? or maybe `compositeWord`?

Reply:
thought about this but since we've been using no separators (for example hostlevel,nonrt, rtetols..) I didn't include that. allowing a new rule while it doesn't apply to all the current available topics feels inconsistent, so we will need to update them to fulfill the rule, which is fine but want to confirm if this is the rule we want to proceed with and apply to all topics

Reply:
it's ok to update later. The current state is sadly inconsistent anyway (mostly my fault) so we will need revisions in the future.

Reply:
it is also OK, actually better, to fix all the topics in a consistent way. It's more work, but starting from clean slate is better if we are willing to endure it.

Reply:
got it, I'll add this and address the needed adjustments in another commit/PR